homeassistant/components/sfr_box/sensor.py

```diff
@@ -0,0 +1,182 @@
+"""SFR Box sensor platform."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from sfrbox_api.bridge import SFRBox
+from sfrbox_api.models import DslInfo, SystemInfo
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import SIGNAL_STRENGTH_DECIBELS, UnitOfDataRate
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import DslDataUpdateCoordinator
+
+
+@dataclass
+class SFRBoxSensorMixin:
+    """Mixin for SFR Box sensors."""
+
+    value_fn: Callable[[DslInfo], StateType]
+
+
+@dataclass
+class SFRBoxSensorEntityDescription(SensorEntityDescription, SFRBoxSensorMixin):
+    """Description for SFR Box sensors."""
+
+
+SENSOR_TYPES: tuple[SFRBoxSensorEntityDescription, ...] = (
+    SFRBoxSensorEntityDescription(
+        key="linemode",
+        name="Line mode",
+        has_entity_name=True,
+        value_fn=lambda x: x.linemode,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="counter",
+        name="Counter",
+        has_entity_name=True,
+        value_fn=lambda x: x.counter,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="crc",
+        name="CRC",
+        has_entity_name=True,
+        value_fn=lambda x: x.crc,
+    ),
+    SFRBoxSensorEntityDescription(
+        key="noise_down",
+        name="Noise down",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
```

Comment:
Signal strength sensors should normally be disabled by default.

Reply:
Addressed in #85038